hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBBatchOperation.java

```diff
@@ -141,39 +339,99 @@ private class FamilyCache {
        * When dbValue is a byte[]/{@link ByteBuffer}, it represents a put-op.
        * Otherwise, it represents a delete-op (dbValue is {@link Op#DELETE}).
        */
-      private final Map<Bytes, Object> ops = new HashMap<>();
+      private final Map<Bytes, Integer> opsKeys = new HashMap<>();
+      private final Map<Integer, Operation> batchOps = new HashMap<>();
```

Comment:
add a description for batchOps similar to opsKeys?
```suggestion
      /**
       * Cache of all batch operations to be applied for this family, keyed by their insertion order index.
       * 
       * Each entry maps a unique integer index (representing the order in which the operation was added)
       * to an {@link Operation} object (such as put, delete, or delete range).
       * 
       * This ensures that all database operations are executed in the same order as they were added,
       * which is crucial for correctly handling overlapping or dependent operations, such as delete ranges.
       */
      private final Map<Integer, Operation> batchOps = new HashMap<>();
```